### ISSUE TITLE
Overview of railway connections when zoomed out

### ISSUE THREAD

**Issue by njam** [osmandapp/OsmAnd#17373]
# 🚀 feature request
### Description

Allow to view and explore railway connections when zoomed out, to plan long-distance train travels. This is currently not possible for various reasons (see below).

For example, here's [OpenRailwayMap](https://openrailwaymap.org/):
<img src=https://github.com/osmandapp/OsmAnd/assets/360800/3138bc9b-b5ff-48e2-9f4b-bd919c4225ff height=200>

### Describe the solution you'd like

I think the way to go is to use OpenRailwayMap's approach, which is well established, and works. This means rendering `railway=rail` , and differentiate the importance of a track by `usage=main|branch|...` ([ref](https://wiki.openstreetmap.org/wiki/OpenRailwayMap/Tagging#Tagging_scheme)).

imho these problems need to be addressed:
1. Bring all the relevant data into the offline map.
    - Render `railway=rail` at any zoom level, not only down to 7 ([ref](https://github.com/osmandapp/OsmAnd-resources/blob/145f0bbc48ab6813f7dcc51a9ddb62c50ab77211/obf_creation/rendering_types.xml#LL3980C1-L3980C49)), at least `usage=main`
    - Add the [`usage=`](https://wiki.openstreetmap.org/wiki/Key:usage) tag for `railway=rail`
2. Reduce visibility of streets, so one can recognize railway tracks on the map
    - We could introduce a new "Road style" which renders all roads in white
3. Render `railway=rail` on the map also when zoomed out. Differentiate "highspeed", "main", "branch" etc.
    - Example render config for OsmAnd: [fuzzysolutions/OsmAndRailwayStyles](https://github.com/fuzzysolutions/OsmAndRailwayStyles/blob/master/RailwayInfra.render.xml)
3. Find a way to enable this in the UI
    - How about enabling this rendering style when the user switches to the rendering mode `public_transport`? I guess it also makes sense to reduce the visibility/colorfulness of roads in this mode.

### Describe alternatives you've considered

- Load OpenRailwayMap data in OsmAnd. But this works only "online". See https://wiki.openstreetmap.org/wiki/OpenRailwayMap/API#Usage_in_OsmAnd
- Instead of rendering `railway=rail` one could also render `route=train`, but I think it's less reliable.

---

wdyt?

I'd be willing to work on this feature, if it's desired. But I will need some guidance.

**Comment by spaanse** [comment]
Yes, I definitely want a style to defocus roads, since it does not make sense in some of the profiles.

**Comment by vshcherb** [comment]
Sounds cool, we have information in detailed basemap but we need to have rendering style

**Comment by slavb18** [comment]
How to override low-zoom railway reader optimization and nested styles in custom rendering styles?

### Description:

We are developing a custom, OpenRailwayMap-inspired map style for OsmAnd ([slavb18/OsmAndRailwayStyles](https://github.com/slavb18/OsmAndRailwayStyles)) with the goal of making the railway network highly visible as thick, solid lines when zoomed out (zoom levels 7–10). 

We are using a style that inherits from `default` (`depends="default"`). The stable baseline configuration can be found here: [RailwayInfra.render.xml](https://github.com/slavb18/OsmAndRailwayStyles/blob/master/RailwayInfra.render.xml).

However, we are running into two main engine/parser-related limitations when trying to make railways visible at zoom levels 7–10:

#### 1. Low-Zoom Reader Optimization (`optimize="read"`)
In `default.render.xml`, railways are optimized out at zoom levels 10 and below:
```xml
<case tag="railway" value="rail" maxzoom="10" order="-1" optimize="read"/>
```
Because of `optimize="read"`, the C++ map reader completely skips loading railways from `.obf` files at zooms $\le 10$. 

If a dependent rendering style tries to override this by redefining the order for standard railways at low zooms (e.g. setting `order="200"`), the override is often ignored unless the child style mirrors the exact XML nesting structure of the parent `<order>` section (nested switches). 

* **Question 1:** What is the correct, officially supported way for a dependent child style to override the parent's `optimize="read"` rules and force the map reader to load elements at low zoom levels?

#### 2. Mismatch/Overriding of Nested Case Styles
In `default.render.xml`, the rendering rule for railways is defined as:
```xml
<case minzoom="7" tag="railway" value="rail">
    ...
    <case maxzoom="7" strokeWidth="0.6"/>
    <case maxzoom="8" strokeWidth="0.9"/>
    <case maxzoom="12" strokeWidth="1:1"/>
    ...
</case>
```
If our child style overrides the main railway rule using `<case minzoom="7" tag="railway" value="rail" strokeWidth="4">`, the XML parser merges them, but the parent style's nested cases (like `maxzoom="8" strokeWidth="0.9"`) remain active and take precedence over our child's outer stroke settings. This forces railways to render as extremely thin, invisible lines at zooms 7–8.

To work around this, we have to define completely separate rules (e.g. `minzoom="2" maxzoom="8"`) to prevent merging, which feels hacky and fragile.

* **Question 2:** How can a dependent style cleanly override nested stroke width and color casings in a parent style without copy-pasting the entire railway rendering node?

#### 3. Low-Zoom Railway Data (Zooms 5–6)
As discussed in [#17373](https://github.com/osmandapp/OsmAnd/issues/17373), standard `.obf` maps generated by OsmAnd currently do not contain any `railway=rail` data below zoom level 7.
* **Question 3:** Are there plans to compile major railway tracks (e.g., `usage=main` or `highspeed=yes`) at zoom levels 5 and 6 into the official map packages so they can be rendered when zoomed out?

**Comment by xmd5a2** [comment]
@slavb18 
1. @Andrewschenko possible can help
2. As far as I know there is no way to do that without  code duplication
3. No plans but we'll think about it
